Synchrotron Spectral Power Flux Plot

GloriaRAH opened the pull request:
@phajy I added the synchrotron spectral power flux `P_syn = epsilon*S_syn ~ nu*F(nu)` calculation on the script and made some changes on the documentation. According our previous meeting, I need to test the code and tried to get similar plot as the figure 2 on [Tavecchio et al (2000)](https://ui.adsabs.harvard.edu/abs/2000ApJ...544L..23T/abstract). I got the following result. The shape is already there but I am not sure about the right values on the y-axis. 

![Tavecchio_Spectral_Power_Flux(1)](https://user-images.githubusercontent.com/100141757/168308745-8bb40ed4-407b-449c-8772-0137bf8d8df1.png)
#

GloriaRAH committed:
Added the synchrotron power spectral flux calculation and changing parameters values using Schwartz et al 2000 and Tavecchio et al 2000 papers.
PKS0637-752

GloriaRAH committed:
Update on the last commit

GloriaRAH committed:
Add some changes on the documentation

phajy commented:
Thanks @GloriaRAH. I'll take a look at the code and make some edits as needed.

- In the S_syn function B (in the Dermer 1997 paper) = 1 / beta so I've updated that.
- The luminosity distance should be in cm but the Hubble constant is in km / s / Mpc so I've converted that to cos units. You can check this calculation by looking up the approximate [luminosity distance for PKS 0637-752 on NED](http://ned.ipac.caltech.edu/byname?objname=PKS+0637-752&hconst=67.8&omegam=0.308&omegav=0.692&wmap=4&corr_z=1).
- I've made the blob radius a parameter called "radius".
- I changed H_0 to 50 km/s/Mpc because that is the value Tavecchio et al. (2000) used.

This moves the synchrotron curve into the right ball park but it is still not in agreement with the Tavecchio et al. (2000) paper. We should check all the parameter values we're using and see if we can figure out what the discrepancy is. Also, the parameters for both models in their Fig. 2 should produce the same synchrotron spectrum. So we could test that too.

I'll try to update the code with the changes I've made to this pull request.

phajy committed:
Updated beta and dL calculations

Fixed a bug with B versus beta. Updated dL to use cgs units for Hubble constant. Put blob radius into the list of parameters.

phajy commented on src/DiscJetConnections.jl:
```diff
@@ -47,13 +47,16 @@ export MyParamStruct
-    θ = 6.0
+    "Angle (radians) between the direction of the blob's motion and the direction to the observer"
+    θ = 6.0*pi/180.0
+    "Radius of emitting region"
+    radius = 1.0E22
```
Added radius as a parameter.

phajy commented on src/DiscJetConnections.jl:
```diff
@@ -47,13 +47,16 @@ export MyParamStruct
     Γ = 10.0
-    "Angle between the direction of the blob's motion and the direction to the observer"
-    θ = 6.0
+    "Angle (radians) between the direction of the blob's motion and the direction to the observer"
+    θ = 6.0*pi/180.0
```
Converted angle to radians.

phajy commented on src/DiscJetConnections.jl:
```diff
@@ -47,13 +47,16 @@ export MyParamStruct
+    "Radius of emitting region"
+    radius = 1.0E22
 
     "Hubble parameter"
     ho = 0.67
     "Hublle constant in km s^-1 Mpc^-1"
-    Ho = 100.0*ho
+    # Ho = 100.0*ho
+    Ho = 50.0
```
H_0 value of 50 in the Tavecchio et al. (2000) paper.

phajy commented on src/synchrotron.jl:
```diff
@@ -57,18 +57,18 @@ function S_syn(ϵ, mps)
     # Γ is the Bulk Lorentz Factor
     # Θ is the angle between the direction of the blob's motion and the direction to the observer
+    # B = 1 / β in the Dermer e tal. (1997) paper
     β = sqrt(1.0 - 1.0/mps.Γ^2)
     μ_obs = cos(mps.θ)
-    D = 1 / (mps.Γ*(1-(β * μ_obs)))
+    D = 1.0 / (mps.Γ*(1 - μ_obs/β))
```
This is the definition of \beta but the Dermer et al. (1997) paper used B which is 1/beta. So I've changed to code to be consistent with their equations (i.e., \mu_obs / \beta = \mu_obs * B).

phajy commented on src/synchrotron.jl:
```diff
@@ -57,18 +57,18 @@ function S_syn(ϵ, mps)
-    D = 1 / (mps.Γ*(1-(β * μ_obs)))
+    D = 1.0 / (mps.Γ*(1 - μ_obs/β))
     
     # Luminosity Distance dL(z)
-    dL = (2.0*mps.c / mps.Ho) * (mps.z+1.0 - sqrt(mps.z+1.0))
+    # Convert H_0 km / s / Mpc to cm / s / cm (cgs)
+    dL = (2.0*mps.c / (mps.Ho * 1.0E5 / 3.086E24 )) * (mps.z+1.0 - sqrt(mps.z+1.0))
```
Need to convert H_0 into cgs units. You can check that the luminosity distance is similar to that on NED, for example.

phajy commented on src/synchrotron.jl:
```diff
@@ -57,18 +57,18 @@ function S_syn(ϵ, mps)
-    # R = 10^3 * Rg
-    R = 1.0E22
-    Vb = (4.0/3.0) * pi * R^3
+    Vb = (4.0/3.0) * pi * mps.radius^3
```
Made radius of blob a parameter.

phajy reviewed (approved):
Looks good. Some changes made and commended below.
